BaseClasses.py

```diff
@@ -334,32 +379,42 @@ def find_common_pool(players: Set[int], shared_pool: Set[str]) -> Tuple[
                 continue
 
             new_itempool: List[Item] = []
-            for item_name, item_count in next(iter(common_item_count.values())).items():
+            # The common item counts are the same for each player to start with, so get the counts for the first player.
+            first_player_counts: Dict[str, int] = next(iter(common_item_count.values()))
+            for item_name, item_count in first_player_counts.items():
                 for _ in range(item_count):
                     new_item = group["world"].create_item(item_name)
                     # mangle together all original classification bits
                     new_item.classification |= classifications[item_name]
                     new_itempool.append(new_item)
 
-            region = Region("Menu", group_id, self, "ItemLink")
-            self.regions.append(region)
-            locations = region.locations
+            linked_items: Dict[str, List[List[Item]]] = {item_name: [[] for _ in range(item_count)]
+                                                         for item_name, item_count in first_player_counts.items()}
```

Comment:
As an extension to this PR, it would be possible to start with an empty `List[List[Item]]` and only append to it from within the loop below. Doing so would eliminate the creation of many empty `List[Item]` in the case of item linking non-advancement items.